libs/server/Storage/Functions/ObjectStore/RMWMethods.cs

```diff
@@ -10,268 +12,236 @@ namespace Garnet.server
-
-                        var writer = new RespMemoryWriter(functionsState.respProtocolVersion, ref output.SpanByteAndMemory);
-                        try
-                        {
-                            var ret = customObjectCommand.NeedInitialUpdate(key, ref input, ref writer);
-                            return ret;
-                        }
-                        finally
-                        {
-                            writer.Dispose();
-                        }
-                    }
+                return customObjectCommand.NeedInitialUpdate(key, ref input, ref writer);
+            }
+            finally
+            {
+                writer.Dispose();
             }
         }
 
         /// <inheritdoc />
-        public bool InitialUpdater(ref byte[] key, ref ObjectInput input, ref IGarnetObject value, ref GarnetObjectStoreOutput output, ref RMWInfo rmwInfo, ref RecordInfo recordInfo)
+        public bool InitialUpdater(ref LogRecord logRecord, in RecordSizeInfo sizeInfo, ref ObjectInput input, ref ObjectOutput output, ref RMWInfo rmwInfo)
         {
+            Debug.Assert(!logRecord.Info.HasETag && !logRecord.Info.HasExpiration, "Should not have Expiration or ETag on InitialUpdater log records");
+
             var type = input.header.type;
+            IGarnetObject value;
             if ((byte)type < CustomCommandManager.CustomTypeIdStartOffset)
             {
                 value = GarnetObject.Create(type);
-                value.Operate(ref input, ref output, functionsState.respProtocolVersion, out _);
+                _ = value.Operate(ref input, ref output, functionsState.respProtocolVersion, out _);
+                _ = logRecord.TrySetValueObjectAndPrepareOptionals(value, in sizeInfo);
                 return true;
             }
-            else
-            {
-                Debug.Assert(type != GarnetObjectType.Expire && type != GarnetObjectType.Persist, "Expire and Persist commands should have been handled already by NeedInitialUpdate.");
 
-                var customObjectCommand = GetCustomObjectCommand(ref input, type);
-                value = functionsState.GetCustomObjectFactory((byte)type).Create((byte)type);
+            var customObjectCommand = GetCustomObjectCommand(ref input, type);
+            value = functionsState.GetCustomObjectFactory((byte)type).Create((byte)type);
 
-                var writer = new RespMemoryWriter(functionsState.respProtocolVersion, ref output.SpanByteAndMemory);
-                try
-                {
-                    var result = customObjectCommand.InitialUpdater(key, ref input, value, ref writer, ref rmwInfo);
-                    return result;
-                }
-                finally
-                {
-                    writer.Dispose();
-                }
+            var writer = new RespMemoryWriter(functionsState.respProtocolVersion, ref output.SpanByteAndMemory);
```

Comment:
We should just use `using var writer = ...` here and elsewhere with `RespMemoryWriter`